lib/iris/common/metadata.py

```diff
@@ -89,25 +94,381 @@ class BaseMetadata(metaclass=_NamedTupleMeta):
+            return tuple(keys)
+
+        return _sort_key(self) < _sort_key(other)
+
+    def __ne__(self, other):
+        result = self.__eq__(other)
+        if result is not NotImplemented:
+            result = not result
+
+        return result
+
+    def _combine(self, other):
+        """Perform associated metadata member combination."""
+        if LENIENT(self.combine):
+            print("lenient combine")
+            values = self._combine_lenient(other)
+        else:
+            print("strict combine")
+
+            def func(field):
+                value = getattr(self, field)
+                return value if value == getattr(other, field) else None
+
+            # Note that, we use "_fields" not "_members".
+            values = [func(field) for field in self._fields]
+
+        return values
+
+    def _combine_lenient(self, other):
+        """
+        Perform lenient metadata member combination.
+
+        Args:
+
+        * other (BaseMetadata):
+            The other metadata participating in the lenient combination.
+
+        Returns:
+            A list of combined metadata member values.
+
+        """
+
+        def func(field):
+            left = getattr(self, field)
+            right = getattr(other, field)
+            result = None
+            if field == "units":
+                # Perform "strict" combination for "units".
+                result = left if left == right else None
+            elif self._is_attributes(field, left, right):
+                result = self._combine_lenient_attributes(left, right)
+            else:
+                if left == right:
+                    result = left
+                elif left is None:
+                    result = right
+                elif right is None:
+                    result = left
+            return result
+
+        # Note that, we use "_members" not "_fields".
+        return [func(field) for field in BaseMetadata._members]
+
+    @staticmethod
+    def _combine_lenient_attributes(left, right):
+        """Leniently combine the dictionary members together."""
+        sleft = set(left.items())
+        sright = set(right.items())
+        # Union of common items.
+        common = sleft & sright
+        # Items in sleft different from sright.
+        dsleft = dict(sleft - sright)
+        # Items in sright different from sleft.
+        dsright = dict(sright - sleft)
+        # Union of common item keys with different values.
+        keys = set(dsleft.keys()) & set(dsright.keys())
+        # Remove (in-place) common item keys with different values.
+        [dsleft.pop(key) for key in keys]
+        [dsright.pop(key) for key in keys]
+        # Now bring the result together.
+        result = dict(common)
+        result.update(dsleft)
+        result.update(dsright)
+
+        return result
+
+    def _compare_lenient(self, other):
+        """
+        Support lenient metadata equality for coordinates.
```

Comment:
"for coordinates" sounds like a hangover from the past here, yes ?

Reply:
@pp-mo Yup, I need to tidy up some of the doc-strings, and that's on my TODO list, thanks.

Reply:
@pp-mo Done.